docs/src/contributing_apub_api_outline.md

```diff
@@ -1,380 +1,592 @@
 # Activitypub API outline
 
-**This document is completely outdated and doesn't reflect the current state of Lemmy at all**
-
-- Start with the [reddit API](https://www.reddit.com/dev/api), and find [Activitypub vocab](https://www.w3.org/TR/activitystreams-vocabulary/) to match it.
-
-<!-- toc -->
-
-- [Actors](#actors)
-  * [User / Person](#user--person)
-  * [Community / Group](#community--group)
-- [Objects](#objects)
-  * [Post / Page](#post--page)
-  * [Post Listings / Ordered CollectionPage](#post-listings--ordered-collectionpage)
-  * [Comment / Note](#comment--note)
-  * [Comment Listings / Ordered CollectionPage](#comment-listings--ordered-collectionpage)
-  * [Deleted thing / Tombstone](#deleted-thing--tombstone)
-- [Actions](#actions)
-  * [Comments](#comments)
-    + [Create](#create)
-    + [Delete](#delete)
-    + [Update](#update)
-    + [Read](#read)
-    + [Like](#like)
-    + [Dislike](#dislike)
-  * [Posts](#posts)
-    + [Create](#create-1)
-    + [Delete](#delete-1)
-    + [Update](#update-1)
-    + [Read](#read-1)
-  * [Communities](#communities)
-    + [Create](#create-2)
-    + [Delete](#delete-2)
-    + [Update](#update-2)
-    + [Join](#join)
-    + [Leave](#leave)
-  * [Moderator](#moderator)
-    + [Ban user from community / Block](#ban-user-from-community--block)
-    + [Delete Comment](#delete-comment)
-    + [Invite a moderator](#invite-a-moderator)
-    + [Accept Invitation](#accept-invitation)
-    + [Reject Invitation](#reject-invitation)
-
-<!-- tocstop -->
+This document is targeted at developers who are familiar with the ActivityPub and ActivityStreams protocols. It gives a detailed outline of the actors, objects and activities used by Lemmy.
+
+Before reading this, have a look at our [Federation Overview](contributing_federation_overview.md) to get an idea how Lemmy federation works on a high level.
+
+Lemmy does not yet follow the ActivityPub spec in all regards. For example, we don't set a valid context indicating our context fields. We also ignore fields like `inbox`, `outbox` or `endpoints` for remote actors, and assume that everything is Lemmy. For an overview of deviations, read [#698](https://github.com/LemmyNet/lemmy/issues/698). They will be fixed in the near future.
```

Comment:
Should also mention that Lemmy is currently completely unable to receive different formats, like array vs single value, or object id vs full object.

Reply:
Sounds good.

Reply:
Done.